app/helpers/application_helper.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module ApplicationHelper
+  def logged_in?
+    !current_user.nil?
+  end
+
+  def log_in(user)
+    session[:user_id] = user.id
+    user.create_new_token
+    cookies.permanent[:remember_token] = user.remember_digest
```

Comment:
What is the value of `cookies.permanent[:remember_token]` after this line gets executed? This is incorrect.

Try for yourself by inserting `byebug` on line 12:

```
  def log_in(user)
    session[:user_id] = user.id
    user.create_new_token
    cookies.permanent[:remember_token] = user.remember_digest
    byebug
  end
```

Here's the result:

![image](https://user-images.githubusercontent.com/22228397/63218931-5d53b700-c12c-11e9-8bea-00ce20152d47.png)

[**REQUIRED**]

Reply:
It'd be also a good idea to use:

```
# Sets a signed cookie, which prevents users from tampering with its value.
# It can be read using the signed method `cookies.signed[:name]`
cookies.signed[:user_id] = current_user.id
```

Read about it [here](https://apidock.com/rails/ActionDispatch/Cookies).